### ISSUE TITLE
error: unknown option --noskipreview

### ISSUE THREAD

**Issue by theslimshaney** [Morganamilo/paru#271]
### Affected Version
paru v1.3.0 - libalpm v12.0.2

### Description
a6c8511 added `--noskipreview` arg
```sh
$ paru --help | grep '\--noskipreview'
    --noskipreview         Don't skip the review process
```
But it doesn't work. skipreview seems to work.

### Output
```sh
$ paru --noskipreview
error: unknown option --noskipreview
```

**Comment by Morganamilo** [comment]
Flag is --review. Typo in help menu.

**Comment by theslimshaney** [comment]
Gotcha. In that case #272 will fix

But it doesn't seem like that actually prevents skipping of review

```
paru --review -Syu
:: Synchronizing package databases...
...
Aur (2) ...

:: Proceed to review? [Y/n]: Y

:: Downloading PKGBUILDs...
 PKGBUILDs up to date
 nothing new to review
resolving dependencies...
looking for conflicting packages...
...
:: Proceed with installation? [Y/n]
```

**Comment by Morganamilo** [comment]
> nothing new to review

There was nothing to review.

**Comment by theslimshaney** [comment]
There was nothing **new** to review, but there were 2 AUR packages (sorry I truncated them from the example) needing upgraded. I wanted to review those but passing --review didn't stop the review from getting skipped.

**Comment by theslimshaney** [comment]
If this is a matter of misinderstanding the flag, I could open a new issue to add something like `--always-review` to include cases where paru thinks the PKGBUILD is already reviewed but the user still wants to review them anyways?

**Comment by Morganamilo** [comment]
Nothing new to review means you have read the pkgbuild at some point and confirmed it's good. There's no need to see the pkgbuild again really.

**Comment by dmi-dmi-dmi** [comment]
@Morganamilo wouldn't it be better to ask user if they still want to perform a review in such cases? In the end user has explicitly passed `--noskipreview` flag to `paru`  and this behavior seems to be very confusing.

**Comment by Morganamilo** [comment]
And then what? The review is a diff from where you last reviewed to the newest commit. What would even be printed?

**Comment by dmi-dmi-dmi** [comment]
Entire PKGBUILD file.

I wanted to review build files again when I did `paru -S pkgname --noskipreview` because initially I edited PKGBUILD for `pkgname` when installed it a month ago and I don't remember what exactly was changed (I forgot to commit my changes right away and `CleanAfter` option wiped them later).

Of course, I can always do `paru -G pkgname` but it seems that current behavior confuses other users too.

**Comment by theslimshaney** [comment]
An option to review the entire PKGBUILD before install would be great, but I was also thinking that there should be a way to "re-review", such as situations where you are reviewing a package and then press a wrong button and paru thinks that you're all done reviewing and never want to review again until next update. Or another similar situation where you review something but then don't end up installing and then you want to review it again later.

**Comment by Morganamilo** [comment]
> Or another similar situation where you review something but then don't end up installing and then you want to review it again later.

In that case, you already reviewed it. It doesn't mater if you installed it or not.

Anyway if you want to see the whole pkgbuild just use --fm mode.

**Comment by dmi-dmi-dmi** [comment]
> In that case, you already reviewed it. It doesn't mater if you installed it or not.

Well, it is hard to argue with you since you constantly change reasoning for paru design decision. For example, earlier you said 

> Nothing new to review means you have read the pkgbuild at some point and confirmed it's good.

I do understand that building and installing a package means confirming that PKGBUILD is good but how on Earth someone would consider not-installing a package to have the same meaning

**Comment by Morganamilo** [comment]
After you review paru will prompt if you want to continue with the install. If you answer yes to this the pkgbuild is marked as reviewed. It doesn't matter if the pkgbuild was not ultimately installed.

**Comment by theslimshaney** [comment]
Let me clarify it then... and then you want to review it again later because you forgot what you reviewed, or you didn't actually review it, or it was a month ago like dmi mentioned, etc. Even if you say yes to continue with install, you can still cancel the install and go back later. Sure paru thinks you already reviewed it cause you said yes to install, but sometimes people will want to review it again.

I think the ideal way would be with the `--review` flag (or perhaps more ideally to some, a new flag `--alwaysreview`) will:
1. Show last reviewed to the newest commit
2. If "last reviewed to the newest commit" is all caught up (such as when you say yes to install, then cancel the install before it finishes, then do another install), then show the previous "last reviewed to the newest commit". This will help cases when you type the wrong command and paru thinks you have reviewed the file but you either haven't or will want to review it again later.

**Comment by Morganamilo** [comment]
> Let me clarify it then... and then you want to review it again later because you forgot what you reviewed, or you didn't actually review it, or it was a month ago like dmi mentioned, etc. Even if you say yes to continue with install, you can still cancel the install and go back later. Sure paru thinks you already reviewed it cause you said yes to install, but sometimes people will want to review it again.

I don't really get this. You did review it. It doesn't matter if you forgot or it's a month later. You reviewed it before, you know it's safe/

>  think the ideal way would be with the --review flag (or perhaps more ideally to some, a new flag --alwaysreview) will:

May as well just use fm mode for that.

**Comment by theslimshaney** [comment]
yeah fm mode might be helpful, I'll check it out.

You may not understand why, and you don't need to understand why. But at least but please just realize that there are cases where the packages can get marked as reviewed where the user does not wish to have them marked as reviewed. Your logic sounds an awful lot like reasons to get rid of undo buttons or backspace key.

This is your tool so I understand if you don't want to add a feature. But if the only reason for not adding the feature is because you don't understand how said feature is actually helpful, please reconsider. 

Either way, thanks for all the work you put into this tool :+1:

**Comment by dmi-dmi-dmi** [comment]
Anyway, seeing this is confusing to me. 

> $ paru --review -S visual-studio-code-insiders-bin  
> :: Resolving dependencies...
> :: Calculating conflicts...
> :: Calculating inner conflicts...
> 
> Aur (1) visual-studio-code-insiders-bin-1610342919-1
> 
> :: Proceed to review? [Y/n]: 
> 
> :: Downloading PKGBUILDs...
>  PKGBUILDs up to date
>  nothing new to review
> fetching devel info...
> ==> Making package: visual-studio-code-insiders-bin 1610342919-1 (Вс 07 мар 2021 22:25:00)
> ==> Retrieving sources...
>   -> Found visual-studio-code-insiders.desktop
>   -> Found visual-studio-code-insiders-url-handler.desktop
>   -> Downloading code_x64_1610342919.tar.gz...
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
>    ....

I didn't even have a chance see PKGBUILD contents prior to executing some code in the build phase. Yet I wanted to see PKGBUILD because I *explicitly* asked `paru` to perform a review phase.

Good software should not make decision on user behalf. It might recommend best course of action but it should always ask for consent unless user explicitly informs that they are fine with any software decisions.

**Comment by dmyTRUEk** [comment]
Two and a half years later I encountered the same issue: I "reviewed" PKGBUILD but not fully, so now I want to "rereview" it, but I have no option to do that? Explicit `--review` flag doing nothing (as opposed to "rereviewing") looks like an unintuitive design to me.

Here is at least three situations in which "rereview" could be useful:
1. The review was done long ago. Maybe then I didn't install a package because I didn't like it, but now maybe I changed my mind.
2. The review wasn't done fully.
3. There is some mistake in it, so building failed, so I want to see what could be wrong with it and maybe fix it.

Maybe the solution would be to mark PKGBUILD as reviewed only after successful installation? Or at least make this behavior as an (experimental?) option (cli arg and config)?
